### ISSUE TITLE
LLVM 4.0 fails to install on Linux: file RPATH_CHANGE could not write new RPATH

### ISSUE THREAD

**Issue by jppelteret** [LLNL/spack#3791]
Dear all,

I've just updated `spack` to 808e9b46493495dff64d9964631faa929bbe4ebd in order to build `LLVM` on Linux (Mint 17.3 based off Ubuntu 14.04 I believe). 

This is how I have my `packages.yaml` setup in order to install LLVM with shared libraries (and minus a couple of troublesome variants)
```
  llvm:
    version: [4.0.0]
    variants: +all_targets+clang+compiler-rt~debug+gold+internal_unwind+libcxx~link_dylib~lldb~polly+shared_libs
```

Unfortunately `spack install llvm` fails upon installation (not compilation) with the following message:
```
-- Installing: <path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/llvm-4.0.0-xve6a26cyhefnycgunchb22fohcs73v4/lib/libc++.so
-- Set runtime path of "<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/llvm-4.0.0-xve6a26cyhefnycgunchb22fohcs73v4/lib/libc++.so.1.0" to "<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/llvm-4.0.0-xve6a26cyhefnycgunchb22fohcs73v4/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/llvm-4.0.0-xve6a26cyhefnycgunchb22fohcs73v4/lib64:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/binutils-2.28-ehmg5z5tvnxbwttzy2hptemrstwyz3rs/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/gettext-0.19.8.1-wu5ducrwcvx3ms2uzrgnl3kvl3vds2hh/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/bzip2-1.0.6-syidqu3x4z355sx5af5ny7okyuhed53q/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/libxml2-2.9.4-qhrsnhlkmv2d26xnmz777fsliwqepbd5/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/xz-5.2.3-x5bmsfrloulpquxn23qh25uvmvnpbovy/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/zlib-1.2.11-q4s2ymg5cupvjutwgdkiklljvsyfhrpf/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/ncurses-6.0-u5zebzkwkfpr65ydde2h4yfrssz453tq/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/python-2.7.13-nbkirwtun5fdceyssv52gglfz3eeuep7/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/openssl-1.1.0e-g6qc24jay4atzhro4d4hxq62toinqhqq/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/readline-7.0-jjz5frcccfgzsfnrhdaut3ftqll5ubpg/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/sqlite-3.8.5-rf2ejcbkj7o5vjjcratrv4jjymrxoezs/lib"
CMake Error at projects/libcxx/lib/cmake_install.cmake:60 (file):
  file RPATH_CHANGE could not write new RPATH:

    <path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/llvm-4.0.0-xve6a26cyhefnycgunchb22fohcs73v4/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/llvm-4.0.0-xve6a26cyhefnycgunchb22fohcs73v4/lib64:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/binutils-2.28-ehmg5z5tvnxbwttzy2hptemrstwyz3rs/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/gettext-0.19.8.1-wu5ducrwcvx3ms2uzrgnl3kvl3vds2hh/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/bzip2-1.0.6-syidqu3x4z355sx5af5ny7okyuhed53q/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/libxml2-2.9.4-qhrsnhlkmv2d26xnmz777fsliwqepbd5/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/xz-5.2.3-x5bmsfrloulpquxn23qh25uvmvnpbovy/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/zlib-1.2.11-q4s2ymg5cupvjutwgdkiklljvsyfhrpf/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/ncurses-6.0-u5zebzkwkfpr65ydde2h4yfrssz453tq/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/python-2.7.13-nbkirwtun5fdceyssv52gglfz3eeuep7/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/openssl-1.1.0e-g6qc24jay4atzhro4d4hxq62toinqhqq/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/readline-7.0-jjz5frcccfgzsfnrhdaut3ftqll5ubpg/lib:<path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/sqlite-3.8.5-rf2ejcbkj7o5vjjcratrv4jjymrxoezs/lib

  to the file:

    <path>/spack/opt/spack/linux-ubuntu14-x86_64/gcc-4.9.4/llvm-4.0.0-xve6a26cyhefnycgunchb22fohcs73v4/lib/libc++.so

  No valid ELF RPATH or RUNPATH entry exists in the file; File does not have
  a valid ELF identification.
Call Stack (most recent call first):
  projects/libcxx/cmake_install.cmake:46 (include)
  projects/cmake_install.cmake:40 (include)
  cmake_install.cmake:61 (include)


make: *** [install] Error 
```
[Build log attached.](https://github.com/LLNL/spack/files/912998/spack-build.out.txt)

In contrast, this configuration builds, installs and works fine on `MacOS`. I've also tried `LLVM 3.8.0 -- 3.9.1`, but these all fail during compilation.

Does anyone have any advice on what I could try to fix this? I've found a [couple](https://cmake.org/pipermail/cmake/2013-September/055755.html) of [posts](http://lists.llvm.org/pipermail/llvm-dev/2014-February/070232.html) with a similar error message, but I'm not sure that any of the information therein is relevant.

**Comment by jcftang** [comment]
JFTR llvm@3.9.1 on centos works fine for me however llvm@4.0.0 on centos7 fails for me with the above error.

**Comment by adamjstewart** [comment]
Closed by #3904.
